apps/dgst.c

```diff
@@ -413,7 +413,16 @@ int dgst_main(int argc, char **argv)
     return ret;
 }
 
-static const char *newline_escape_filename(const char *file)
+/*
+ * The newline_escape_filename function performs newline escaping for any 
+ * filename that contains a newline.  This function also takes a pointer 
+ * to backslash. The backslash pointer is a flag to indicating whether a newline
+ * is present in the filename.  If a newline is present, the backslash flag is 
+ * set and the output format will contain a backslash at the beginning of the
+ * digest output. This output format is to replicate the output format found
+ * in *sum and sha256sum. This aims to preserve backwards compatibility.
```

Comment:
If you mention `*sum`, there's no need to mention `sha256sum` specifically.  However, to avoid confusion, you could consider ending the sentence like this:

``` C
 * in the '*sum' checksum program. ...
```

Reply:
Good point.  I have altered this in my new commit.